commons/protos/policy_interface.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+
+message Graph {
+  message Node {
+    string version = 1;
+    string image = 2;
+    map<string, string> metadata = 3;
+  }
+
+  message Edge {
+    uint64 from = 1;
+    uint64 to = 2;
+  }
+
+  repeated Node nodes = 1;
+  repeated Edge edges = 2;
+}
+
+message PluginInput {
+  Graph graph = 1;
+  map<string, string> parameters = 2;
+}
+
+message PluginOutput {
+  message Error {
+    enum Status {
+      SUCCESS = 0;
+      FAILURE = 1;
+    };
+
+    Status plugin_status = 1;
+    string description = 2;
+    uint64 http_status = 3;
+  }
+
+  Error error = 1;
```

Comment:
I'm not a protobuf expert, but would it be possible to model the `Result<GraphAndParams, Error>` more directly? I seem to remember `oneof` would allow that. I think we can even re-use (after renaming) the graph+params message we use for input.